python/pyspark/sql/pandas/_typing/__init__.pyi

```diff
@@ -348,10 +349,22 @@ PandasCogroupedMapFunction = Union[
 ArrowGroupedMapFunction = Union[
     Callable[[pyarrow.Table], pyarrow.Table],
     Callable[[Tuple[pyarrow.Scalar, ...], pyarrow.Table], pyarrow.Table],
+    Callable[[Iterator[pyarrow.RecordBatch]], Iterator[pyarrow.RecordBatch]],
```

Comment:
what about adding a new `ArrowGroupedMapIterFunction` for the new signature?

Reply:
Sure I can do that, and then just Union them in the function signature directly? Or create like `ArrowGroupedMapTableFunction` and `ArrowGroupedMapIterFunction` and then union them to `ArrowGroupedMapFunction`?

Reply:
I thought we need a new `PythonEvalType` and Exec node for this new signature.
If we don't need them, then existing change is fine.